dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java

```diff
@@ -209,6 +219,51 @@ public void unexport() {
         repository.unregisterProvider(providerModel);
     }
 
+    private void waitForIdle() {
```

Comment:
Graceful offline also needs to consider that the connection is no longer receiving new requests, and the thread pool has completed all tasks within the specified time. Is it possible to check the number of tasks in some thread pools here?

Reply:
Concurrent check can be finished seperately. This PR is focus on wait for consumer not to use the offlined provider.